curp/src/client_new/unary.rs

```diff
@@ -0,0 +1,620 @@
+            .await??;
+        Ok(())
+    }
+
+    /// Send propose to publish a node id and name
+    async fn propose_publish(
+        &self,
+        node_id: ServerId,
+        node_name: String,
+    ) -> Result<(), Self::Error> {
+        let propose_id = self.gen_propose_id().await?;
+        let req = PublishRequest::new(propose_id, node_id, node_name);
+        let timeout = self.config.wait_synced_timeout;
+        let _ig = self
+            .for_leader(|conn| async move { conn.publish(req, timeout).await })
+            .await??;
+        Ok(())
+    }
+
+    /// Send fetch read state from leader
+    async fn fetch_read_state(&self, cmd: &C) -> Result<ReadState, CurpError> {
+        // Same as fast_round, we blame the serializing error to the server even
+        // thought it is the local error
+        let req = FetchReadStateRequest::new(cmd, self.cluster_version()).map_err(|ser_err| {
+            warn!("serializing error: {ser_err}");
+            CurpError::from(ser_err)
+        })?;
+        let timeout = self.config.wait_synced_timeout;
+        let state = self
+            .for_leader(|conn| async move { conn.fetch_read_state(req, timeout).await })
+            .await??
+            .into_inner()
+            .read_state
+            .unwrap_or_else(|| unreachable!("read_state must be set in fetch read state response"));
+        Ok(state)
+    }
+
+    /// Send fetch cluster requests to all servers
+    /// Note: The fetched cluster may still be outdated if `linearizable` is false
+    async fn fetch_cluster(&self, linearizable: bool) -> Result<FetchClusterResponse, CurpError> {
+        let timeout = self.config.wait_synced_timeout;
+        if !linearizable {
+            // firstly, try to fetch the local server
+            if let Some(connect) = <Unary<C> as ClientApi>::local_connect(self) {
+                /// local timeout, in fact, local connect should only be bypassed, so the timeout maybe unused.
+                const FETCH_LOCAL_TIMEOUT: Duration = Duration::from_secs(1);
+
+                let resp = connect
+                    .fetch_cluster(FetchClusterRequest::default(), FETCH_LOCAL_TIMEOUT)
+                    .await
+                    .unwrap_or_else(|e| {
+                        unreachable!(
+                            "fetch cluster from local connect should never failed, err {e:?}"
+                        )
+                    })
+                    .into_inner();
+                return Ok(resp);
+            }
+        }
+        // then fetch the whole cluster
+        let (quorum, mut responses) = self.for_each_server(|conn| async move {
+            (
+                conn.id(),
+                conn.fetch_cluster(FetchClusterRequest { linearizable }, timeout)
+                    .await
+                    .map(Response::into_inner),
+            )
+        });
+        let majority_quorum = majority_quorum(quorum);
```

Comment:
Same as above. Majority and quorum means the same thing I think.

Reply:
Same as above

Reply:
I think they mean the same thing too.